bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ErrorUtils.java

```diff
@@ -99,38 +111,39 @@ public static ErrorValue createCancelledFutureError() {
         return (ErrorValue) createError(BallerinaErrorReasons.FUTURE_CANCELLED);
     }
 
-    public static BError createTypeCastError(Object sourceVal, Type targetType) {
-        throw createError(BallerinaErrorReasons.TYPE_CAST_ERROR,
-                          BLangExceptionHelper.getErrorMessage(RuntimeErrors.TYPE_CAST_ERROR,
-                                                               TypeChecker.getType(sourceVal), targetType));
-
+    public static BError createRuntimeError(Module module, RuntimeErrorType errorType, BMap<BString, Object> detail) {
+        return createRuntimeError(module, errorType, null, detail);
     }
 
-    public static BError createBToJTypeCastError(Object sourceVal, String targetType) {
-        throw createError(BallerinaErrorReasons.TYPE_CAST_ERROR,
-                          BLangExceptionHelper.getErrorMessage(RuntimeErrors.J_TYPE_CAST_ERROR,
-                                                               TypeChecker.getType(sourceVal), targetType));
+    public static BError createRuntimeError(Module module, RuntimeErrorType errorType, BError cause,
+                                            BMap<BString, Object> detail) {
+        try {
+            return createError(module, errorType.getErrorName(), getModulePrefixedErrorName(module, errorType),
+                    cause, detail);
+        } catch (BError e) {
+            throw ErrorCreator.createError(StringUtils.fromString(e.getMessage()));
+        }
     }
 
-    public static BError createNumericConversionError(Object inputValue, Type targetType) {
-        throw createError(BallerinaErrorReasons.NUMBER_CONVERSION_ERROR,
-                          BLangExceptionHelper.getErrorMessage(
-                                  RuntimeErrors.INCOMPATIBLE_SIMPLE_TYPE_CONVERT_OPERATION,
-                                  TypeChecker.getType(inputValue), inputValue, targetType));
+    public static BError getRuntimeError(Module module, RuntimeErrorType errorType, Object... params) {
```

Comment:
Shall we change this method name to createRuntimeError for consistency with other methods?